src/mongocxx/test/collection.cpp

```diff
@@ -2665,4 +2667,56 @@ TEST_CASE("bulk_write with container", "[collection]") {
     REQUIRE(result->inserted_count() == 10);
     REQUIRE(collection.count_documents({}) == 10);
 }
+
+/* Regression test for CXX-2028. */
+TEST_CASE("find_and_x operations append write concern correctly", "[collection]") {
+    instance::current();
+    mongocxx::client client{uri{}};
+    mongocxx::write_concern wc;
+    wc.acknowledge_level(mongocxx::write_concern::level::k_acknowledged);
+
+    mongocxx::collection collection = client["fam_wc"]["collection"];
+    collection.drop();
+    collection.insert_one(make_document(kvp("x", 1)));
+
+    stdx::optional<bsoncxx::document::value> doc;
+    /* 4.4. servers will reply with an error, causing an exception. */
+    /* find_one_and_update */
+    mongocxx::options::find_one_and_update find_one_and_update_opts;
+    find_one_and_update_opts.write_concern(wc);
+    doc = collection.find_one_and_update(make_document(),
```

Comment:
you should be able to just pass {} as the first parameter here as an empty filter, rather than having to call make_document()

`collection.find_one_and_replace({}, ...);`

Reply:
Oh right, done.